### PR TITLE
fix string to JSON conversion and update prettier version

### PR DESCRIPTION
**Pull request by gadielkalleb**
## FIX [issue-81](https://github.com/francescov1/mongoose-tsgen/issues/81)

* fix string to JSON conversion in src/helpers/tsReader.ts
* update prettier version - this error happens at the end of the execution after the typing file was already created when it was going to run prettier as described in this [forum](https://giters.com/prettier/prettier/issues/11533)

### CODE REVIEW

File: src/helpers/tsReader.ts [hunk 1]
```diff
@@ -372,7 +372,11 @@ export const registerUserTs = (basePath: string): (() => void) | null => {
 
   // handle path aliases
   const tsConfigString = fs.readFileSync(foundPath, "utf8");
-  const tsConfig = JSON.parse(stripJsonComments(tsConfigString));
+
+  const tsConfig = stripJsonComments(tsConfigString) as unknown as { [key: string]: any };
```
Comment (francescov1) [inline src/helpers/tsReader.ts:376]: This `JSON.parse` is required to parse the user's TSConfig file and register the `paths` option. The `stripJsonComments` returns a string, so this cannot simply be casted to `{ [key: string]: any };`.

Parsing a tsconfig.json file with `JSON.parse` should not throw errors unless the file is invalid. A better alternative in my opinion is to wrap the `JSON.parse` in a try/catch and give a specific error message, eg:

```typescript
try {
   const tsConfig = JSON.parse(stripJsonComments(tsConfigString));
} catch(err) {
  throw new Error("Error parsing your tsconfig.json file, please ensure the format is valid");
}
```

File: src/index.ts [hunk 2]
```diff
@@ -57,30 +57,34 @@ class MongooseTsgen extends Command {
...
+
+      return {
+        flags: {
+          ...configFileFlags,
+          ...cliFlags
+        } as FlagConfig,
+        args
+      };
+    } catch (error: any) {
+      throw new Error(error.message);
```
Comment (francescov1) [inline src/index.ts:86]: What's the purpose for this change?

**Comment by francescov1** [comment]
Thanks for the PR. Added a few comments!

File: src/index.ts [hunk 3]
```diff
@@ -136,12 +140,15 @@ class MongooseTsgen extends Command {
...
         generator.saveFile({ genFilePath, sourceFile });
+
         if (!flags["no-format"]) await formatter.format([genFilePath]);
         this.log("Writing complete 🐒");
         process.exit();
       }
     } catch (error) {
-      this.error(error as Error);
+      const err = error as Error;
+      this.log(`❌ - ${err.stack}`);
```
Comment (francescov1) [inline src/index.ts:150]: This should only be logged in debug mode, eg:

```
if (flags.debug) {
  this.log(err.stack)
}
```

File: src/helpers/tsReader.ts (hunk 1 above)
Comment (gadielkalleb) [reply src/helpers/tsReader.ts:376]: Strange this behavior that you described is not happening when running this project currently when we install or clone it.

locally I needed to remove JSON.parser and everything worked.

I was curious wondering if it was my machine or operating system.

I did the same tests that I described in the issue and have different operating systems (macOs, ubuntu and windows 10), they all gave the same error. even using the example in the README.

today this lib runs locally in your environment without having this JSON.parser problem?

File: src/index.ts (hunk 2 above)
Comment (gadielkalleb) [reply src/index.ts:86]: I made this change because error.stack did not appear (even declaring the **-d** option)

I had to add this catch to catch the error.

without that I only had the generic information

```SyntaxError: Unexpected token } in JSON at position 683```

not knowing where was the problem with parsing the JSON

File: src/index.ts (hunk 3 above)
Comment (gadielkalleb) [reply src/index.ts:150]: the -d option is not working, I had to do a debugger using the vscode tool to get to the error

File: src/helpers/tsReader.ts (hunk 1 above)
Comment (francescov1) [reply src/helpers/tsReader.ts]: Yep we've been using this lib in production at my company for a few months without issues. 

Are you sure your `tsconfig.json` file is correct? If so, then JSON.parse shouldn't have any issues. All the code is doing is reading this file and then parsing it. Can you try printing out the output from `stripJsonComments(tsConfigString)` and post here? Will help quite a bit

**Comment by gadielkalleb** [comment]
@francescov1 I did some more tests here, it seems that everything is working, maybe it was just the error handling and prettier version update.

Is there an item that you think is interesting to continue our review?

**Comment by francescov1** [comment]
Ah interesting, gotta love programming 😅. Glad to hear it's resolved! I'll go ahead and merge this in, thanks for the help!
